docs/required_libraries.md

```diff
@@ -0,0 +1,47 @@
+# Required libraries
+
+This is a list of known third-party libraries required by the Crystal compiler and the standard library.
+
+## Runtime dependencies
+
+The following libraries are required by Crystal's runtime, and must be present when building or running any program, unless the `--prelude=empty` compiler option is passed.
+
+| Library | Description | License |
+|---------|-------------|---------|
+| [PCRE][libpcre] | Perl Compatible Regular Expressions. Implements [`Regex`](https://crystal-lang.org/api/Regex.html). | [BSD](http://www.pcre.org/licence.txt)
+| [Boehm GC][libgc] | The Boehm-Demers-Weiser conservative garbage collector. Performs automatic memory management. | [MIT-style](https://github.com/ivmai/bdwgc/blob/master/LICENSE)
+| [Libevent][libevent] | An event notification library. Implements concurrency features such as [`Fiber`](https://crystal-lang.org/api/Fiber.html) and the event loop. | [Modified BSD](https://github.com/libevent/libevent/blob/master/LICENSE)
+| [compiler-rt builtins][compiler-rt] | Provides optimized implementations for low-level routines required by code generation, such as integer multiplication. Several of these routines are ported to Crystal directly. | [MIT / UIUC][compiler-rt]
```

Comment:
Where are we actually still using compiler-rt builtins? Is there a way to find out?